docs/reference/schemas/config/functions/tryIndexFromEnd.md

```diff
@@ -0,0 +1,291 @@
+counting backward from the end using a one-based index. Unlike standard array
+indexing that might fail with out-of-bounds errors, this function returns `null`
+when the index is invalid or out of range.
+
+This is particularly useful when working with dynamic arrays where the length
+isn't known in advance, or when implementing fallback logic that needs to handle
+missing data gracefully. The function uses a one-based index, meaning `1`
+refers to the last element, `2` to the second-to-last, and so on.
+
+The function returns `null` in the following cases:
+
+- The reverse index is greater than the array length
+- The reverse index is zero or negative
+- The array is empty
+
+## Examples
+
+### Example 1 - Access recent deployment history safely
+
+Use `tryIndexFromEnd()` to access recent deployment records when you're not
+certain how many deployments have occurred. This is useful for rollback
+scenarios where you want to retrieve the previous deployment without causing
+errors if the history is empty or shorter than expected. This example uses
+[`createArray()`][05] to build the deployment history and [`last()`][00] to get
+the current deployment.
+
+```yaml
+# tryIndexFromEnd.example.1.dsc.config.yaml
+$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+resources:
+- name: Deployment Rollback
+  type: Microsoft.DSC.Debug/Echo
+  properties:
+    output:
+      currentDeployment: "[last(createArray('v1.0.0', 'v1.1.0', 'v1.2.0'))]"
+      previousDeployment: "[tryIndexFromEnd(createArray('v1.0.0', 'v1.1.0', 'v1.2.0'), 2)]"
+      fallbackDeployment: "[tryIndexFromEnd(createArray('v1.0.0', 'v1.1.0', 'v1.2.0'), 10)]"
+```
+
+```bash
+dsc config get --file tryIndexFromEnd.example.1.dsc.config.yaml
+```
+
+```yaml
+results:
+- name: Deployment Rollback
+  type: Microsoft.DSC.Debug/Echo
+  result:
+    actualState:
+      output:
+        currentDeployment: v1.2.0
+        previousDeployment: v1.1.0
+        fallbackDeployment: null
+messages: []
+hadErrors: false
+```
+
+The function returns `v1.1.0` for the second-to-last deployment, and `null` for
+the non-existent 10th-from-last deployment, allowing your configuration to
+handle missing data gracefully.
+
+### Example 2 - Select backup retention with safe defaults
+
+Use `tryIndexFromEnd()` to implement flexible backup retention policies that
+adapt to available backups without failing when fewer backups exist than
+expected. This example retrieves the third-most-recent backup if available. This
+example uses [`createArray()`][05] to build the backup timestamps.
+
+```yaml
+# tryIndexFromEnd.example.2.dsc.config.yaml
+$schema: https://aka.ms/dsc/schemas/v3/bundled/config/document.json
+resources:
+- name: Backup Retention
+  type: Microsoft.DSC.Debug/Echo
+  properties:
+    output:
+      backups: "[createArray(20250101, 20250108, 20250115, 20250122, 20250129)]"
+      retainAfter: "[tryIndexFromEnd(createArray(20250101, 20250108, 20250115, 20250122, 20250129), 3)]"
```

Comment:
Rather than repeating the array, would it be better to have it as a parameter default value?  Or a variable?